textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -41,6 +48,11 @@
     private ScoreDoc[] scoreDocs;
     private IndexReader luceneIndexReader;
     private DataReaderPredicate dataReaderPredicate;
+    private ArrayList<String> queryTokens;

```

Comment:
There variable are created only once, when calling the open() function. If i move them inside getNextTuple() they will be created for every function call.